src/webapp/gradebookServer.js

```diff
@@ -160,11 +160,23 @@ app.get('/years', function(request, response) {
       passwordText, request.query.host, request.query.port);
 
    //Get the params from the url
-   var instructorID = request.query.instructorid;
+   var userID = request.query.userID;
+   var userRole = request.query.userRole; //assuming this is provided at the UI level
+   var queryText;
 
    //Set the query text
-   var queryText = 'SELECT Year FROM gradebook.getInstructorYears($1);';
-   var queryParams = [instructorID];
+   if (userRole == 'alpha_GB_Instructor')
+   {
+      queryText = 'SELECT Year FROM getInstructorYears($1);';
```

Comment:
```suggestion
      queryText = 'SELECT Year FROM getInstructorYears($1);';
      queryParams = [userID];
```

Sets parameters to `queryParams` for the one query that needs it.